catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -13,200 +13,798 @@
+                        "ValidationError: Metacard does not have a registry id.");
+            }
+
+            Set<String> tags = metacard.getTags();
+            if (!tags.contains(RegistryConstants.REGISTRY_TAG)) {
+                throw new FederationAdminException(
+                        "ValidationError: Metacard does not have a registry tag.");
+            }
+        }
     }
 
-    @Override
-    public List<RegistryPackageType> getLocalRegistryObjects() throws FederationAdminException {
-        return new ArrayList<>();
+    private Metacard getRegistryIdentityMetacard() throws FederationAdminException {
+        Metacard identityMetacard = null;
+
+        List<Filter> filters = getBasicFilter();
+        filters.add(filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_IDENTITY_NODE)
+                .is()
+                .bool(true));
+        Filter filter = filterBuilder.allOf(filters);
+
+        List<Metacard> identityMetacards = getRegistryMetacardsByFilter(filter);
+        if (CollectionUtils.isNotEmpty(identityMetacards)) {
+
+            if (identityMetacards.size() > 1) {
+                String message =
+                        "Error getting registry identity metacard. More than one result found.";
+                LOGGER.error("{} Found these: {}", message, identityMetacards);
+                throw new FederationAdminException(message);
+            }
+
+            identityMetacard = identityMetacards.get(0);
+        }
+
+        return identityMetacard;
     }
 
-    @Override
-    public List<RegistryPackageType> getRegistryObjects() throws FederationAdminException {
-        return new ArrayList<>();
+    private List<Metacard> getRegistryMetacardsByFilter(Filter filter)
+            throws FederationAdminException {
+        return getRegistryMetacardsByFilter(filter, null);
     }
 
-    @Override
-    public RegistryPackageType getRegistryObjectByMetacardId(String metacardId)
+    private List<Metacard> getRegistryMetacardsByFilter(Filter filter, List<String> sourceIds)
             throws FederationAdminException {
-        return null;
+        if (filter == null) {
+            throw new FederationAdminException(
+                    "Error getting registry metacards. Null filter provided.");
+        }
+        List<Metacard> registryMetacards = new ArrayList<>();
+        PropertyName propertyName = new PropertyNameImpl(Metacard.MODIFIED);
+        SortBy sortBy = new SortByImpl(propertyName, SortOrder.ASCENDING);
+        QueryImpl query = new QueryImpl(filter);
+        query.setSortBy(sortBy);
+        query.setPageSize(PAGE_SIZE);
+        QueryRequest queryRequest = new QueryRequestImpl(query, sourceIds);
+
+        try {
+            QueryResponse queryResponse =
+                    security.runWithSubjectOrElevate(() -> catalogFramework.query(queryRequest));
+            List<Result> results = queryResponse.getResults();
+
+            registryMetacards.addAll(results.stream()
```

Comment:
Why not just `registryMetacards = results.stream()...` and skip the extra copy at the end?